code/datums/components/fearful/sources.dm

```diff
@@ -0,0 +1,165 @@
+// Terror source handlers
+/// Simple source which passively increases terror based on a single condition and can do something when its added/removed
+/datum/terror_handler/simple_source
+	handler_type = TERROR_HANDLER_SOURCE
+	/// How much terror is added per second
+	var/buildup_per_second = 10
+	/// Have we already applied our effects?
+	var/active = FALSE
+
+/datum/terror_handler/simple_source/tick(seconds_per_tick, terror_buildup)
+	. = ..()
+	if (check_condition(seconds_per_tick, terror_buildup) && terror_buildup < TERROR_BUILDUP_PASSIVE_MAXIMUM)
+		if (!active)
+			on_activation(terror_buildup)
+		. += min(buildup_per_second * seconds_per_tick, TERROR_BUILDUP_PASSIVE_MAXIMUM - terror_buildup)
+	else if (active)
+		on_deactivation(terror_buildup)
+
+/// Proc that children should override with their conditions
+/datum/terror_handler/simple_source/proc/check_condition(seconds_per_tick, terror_buildup)
+	return !HAS_TRAIT(owner, TRAIT_FEARLESS) && !HAS_TRAIT(owner, TRAIT_MIND_TEMPORARILY_GONE) && owner.stat < UNCONSCIOUS
+
+/// Proc that's called when the effect is first applied, for moodlets and alike
+/datum/terror_handler/simple_source/proc/on_activation(terror_buildup)
+	active = TRUE
+
+/// Proc that's called when the effect stops working, for moodlets and alike
+/datum/terror_handler/simple_source/proc/on_deactivation(terror_buildup)
+	active = FALSE
+
+/// Makes the owner terrified of darkness
+/datum/terror_handler/simple_source/nyctophobia
+	buildup_per_second = 5 // Takes about two minutes to reach maximum
+
+/datum/terror_handler/simple_source/nyctophobia/Destroy(force)
+	owner.clear_mood_event("nyctophobia")
+	return ..()
+
+/datum/terror_handler/simple_source/nyctophobia/check_condition(seconds_per_tick, terror_buildup)
+	. = ..()
+	if (!.)
+		return
+
+	if (ishuman(owner))
+		var/mob/living/carbon/human/as_human = owner
+		if(as_human.dna?.species.id in list(SPECIES_SHADOW, SPECIES_NIGHTMARE))
+			return FALSE
+
+	if((owner.sight & SEE_TURFS) == SEE_TURFS)
+		return FALSE
+
+	var/lit_tiles = 0
+	var/unlit_tiles = 0
+
+	for(var/turf/open/turf_to_check in range(1, owner))
+		var/light_amount = turf_to_check.get_lumcount()
+		if(light_amount > LIGHTING_TILE_IS_DARK)
+			lit_tiles++
+		else
+			unlit_tiles++
+
+	return lit_tiles < unlit_tiles
```

Comment:
You changed how nyctophobia to check surrounding tiles that I don't see anywhere in your changelog. Before it only checked the current turf you were on to see if it was dark/lit.

This probably fixes a lot of issues:
- https://github.com/tgstation/tgstation/issues/37492
- https://github.com/tgstation/tgstation/issues/57121
- https://github.com/tgstation/tgstation/issues/39455
- https://github.com/tgstation/tgstation/issues/69684